### PR TITLE
Release/v0.31.0

### PR DESCRIPTION
**Pull request by Liamsi**
- [x] do we need any upgrade instructions (prob. not)
 - [x] double check that the changelog is complete; merged without a changelog entry:
      - https://github.com/tendermint/tendermint/pull/3371 (add shoutout to @silasdavis in entry)
      - https://github.com/tendermint/tendermint/pull/3351
      - https://github.com/tendermint/tendermint/pull/3359
      - https://github.com/tendermint/tendermint/pull/3358
      - https://github.com/tendermint/tendermint/pull/3354
      - https://github.com/tendermint/tendermint/pull/3301
      - https://github.com/tendermint/tendermint/pull/3397
      - ... ?
 - [x] double check we did not forget any external contributor
 - [x] double check versions.go

https://github.com/tendermint/tendermint/blob/5483ac6b0a2a75bc140487e8405392533a3f1687/CHANGELOG.md#v0310

### CODE REVIEW

File: abci/types/types.proto [hunk 1]
```diff
@@ -207,13 +207,13 @@ message ResponseCommit {
 // ConsensusParams contains all consensus-relevant parameters
 // that can be adjusted by the abci app
 message ConsensusParams {
-  BlockSizeParams block_size = 1;
+  BlockParams block = 1;
```
Comment (Liamsi) [inline abci/types/types.proto]: Does this renaming justify a version bump of `ABCIVersion` in version.go?

Comment (ebuchman) [reply abci/types/types.proto]: Good catch. I think this is technically a breaking change in ABCI

**Comment by melekes** [comment]
We should add changelog entries for:
- https://github.com/tendermint/tendermint/pull/3351
- https://github.com/tendermint/tendermint/pull/3359
- https://github.com/tendermint/tendermint/pull/3358 

Released in v0.30.2:
- https://github.com/tendermint/tendermint/pull/3397

**Comment by melekes** [comment]
We need to add the following to UPGRADING.md:

Since the pubsub is no longer blocks on sending, some WS clients might stop working as expected. If your WS client is not consuming events fast enough, Tendermint can terminate the subscription. In this case, the WS client will receive an error with description:

```json
{
  "jsonrpc": "2.0",
  "id": "{ID}#event",
  "error": {
    "code": -32000,
    "msg": "Server error",
    "data": "subscription was cancelled (reason: client is not pulling messages fast enough)" // or "subscription was cancelled (reason: Tendermint exited)"
  }
}
```

**Comment by melekes** [comment]
- [ ] We should actually try and upgrade localnet to see if anything breaks

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+
+* P2P Protocol
+
+### FEATURES:
+- [mempool] [\#3079](https://github.com/tendermint/tendermint/issues/3079) bound mempool memory usage (`mempool.max_txs_bytes` is set to 1GB by default; see config.toml)
+  mempool's current `txs_total_bytes` is exposed via `total_bytes` field in
+  `/num_unconfirmed_txs` and `/unconfirmed_txs` RPC endpoints.
+- [config] [\#2920](https://github.com/tendermint/tendermint/issues/2920) Remove `consensus.blocktime_iota` parameter
+- [genesis] [\#2920](https://github.com/tendermint/tendermint/issues/2920) Add `time_iota_ms` to block's consensus parameters (not exposed to the application)
+- [genesis] [\#2920](https://github.com/tendermint/tendermint/issues/2920) Rename `consensus_params.block_size` to `consensus_params.block`
```
Comment (Liamsi) [inline CHANGELOG.md]: Does this need to go under breaking then?

Comment (melekes) [reply CHANGELOG.md]: Yes

**Comment by Liamsi** [comment]
>  We should actually try and upgrade localnet to see if anything breaks

I ran a local testnet (0.30.0) stopped that, rebuild with the current state in this branch and restarted the localnet (with 0.31.0) It continued making blocks without anything noteworthy in the logs.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,60 @@
 # Changelog
 
+## v0.31.0
+
+*March 13th, 2019*
+
```
Comment (melekes) [inline CHANGELOG.md]: This release brings pubsub 2.0, limits the mempool size to 1GB (`max_txs_bytes`) and number of `/subscribe` WS clients (`max_subscription_clients`) and adds `/unsubscribe_all` endpoint to lite client. It also contains many smaller improvements and bugfixes. Pubsub 2.0 is an improved version of the older pubsub, which is a) non-blocking b) has nicer API. Note our HttpClient's interface got updated to reflect the pubsub changes and now also has a better API for WS subscriptions.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,60 @@
...
+Special thanks to external contributors on this release:
+@danil-lashin, @guagualvcha, @jleni, @siburu, @silasdavis, @srmo, @Stumble, @svenstaro 
+
+### BREAKING CHANGES:
+
+* CLI/RPC/Config
+- [rpc/client] Update Subscribe interface to reflect new pubsub/eventBus API [ADR-33](https://github.com/tendermint/tendermint/blob/develop/docs/architecture/adr-033-pubsub.md)
```
Comment (melekes) [inline CHANGELOG.md]: - [config] Add `rpc.max_subscription_clients` config parameter to control how many unique clientIDs that can /subscribe at the same time
- [config] Add `rpc.max_subscriptions_per_client` config parameter to control how many unique queries a given client can /subscribe to
- [config] Add `rpc.max_subscription_clients` config parameter allowing you to change time  to wait for a tx to be committed during /broadcast_tx_commit

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,93 @@
...
+@danil-lashin, @guagualvcha, @siburu, @silasdavis, @srmo, @Stumble, @svenstaro
+
+This release is primarily about the new pubsub implementation, dubbed `pubsub 2.0`, and related changes,
+like configurable limits on the number of active RPC subscriptions at a time (`max_subscription_clients`).
+Pubsub 2.0 is an improved version of the older pubsub that is non-blocking and has a nicer API.
+Note the improved pubsub API also resulted in some improvements to the HTTPClient interface and the API for WebSocket subscriptions.
+This release also adds a configurable limit to the mempool size, `max_txs_bytes`, with
+default 1GB, and includes many smaller improvements and bug-fixes.
+
+See the [v0.31.0
```
Comment (Liamsi) [inline CHANGELOG.md]: Nit: this line-break looks not intended. Shows up OK though: https://github.com/tendermint/tendermint/blob/551b6322f5a74f578a8487001e805e4e1da6394d/CHANGELOG.md

**Pull request merged**